packages/flutter/lib/src/widgets/text_editing_actions.dart

```diff
@@ -0,0 +1,558 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'framework.dart';
+import 'text_editing_action.dart';
+import 'text_editing_intents.dart';
+
+/// An [Actions] widget that handles the default text editing behavior for
+/// Flutter on the current platform.
+///
+/// This default behavior can be overridden by placing an [Actions] widget lower
+/// in the widget tree than this. See [TextEditingShortcuts] for an example of
+/// remapping keyboard keys to an existing text editing [Intent].
+///
+/// {@tool snippet}
+///
+/// This example shows how to use an additional [Actions] widget to override
+/// the left arrow key [Intent] and make it move the cursor to the right
+/// instead.
+///
+/// ```dart
+/// final TextEditingController controller = TextEditingController(
+///   text: "Try using the keyboard's arrow keys and notice that left moves right.",
+/// );
+///
+/// @override
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Center(
+///       child: Actions(
+///         actions: <Type, Action<Intent>>{
+///           ArrowLeftTextIntent: TextEditingAction<ArrowLeftTextIntent>(
+///             onInvoke: (ArrowLeftTextIntent intent, EditableTextState editableTextState) {
+///               editableTextState.renderEditable.moveSelectionRight(SelectionChangedCause.keyboard);
+///             },
+///           ),
+///         },
+///         child: TextField(
+///           controller: controller,
+///         ),
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [TextEditingShortcuts], which maps keyboard keys to many of the
+///     [Intent]s that are handled here.
+class TextEditingActions extends StatelessWidget {
```

Comment:
Should this have "default" in the name?

Reply:
Yeah you're right, since that's what this is: just Actions with the default text editing actions in it.  I'll prepend "Default" to this and also to TextEditingShortcuts.